### ISSUE TITLE
$let should restrict variables names to `/^[a-zA-Z_][a-zA-Z0-9_]*$/`

### ISSUE THREAD

**Issue by jonasfj** [taskcluster/json-e#159]
In the context we restrict variable names to: `/^[a-zA-Z_][a-zA-Z0-9_]*$/`

We should do the same in $let as we do for `each(...)` too...

This aims to prevent variable names that can't be accessed like `{$let: {" bla bla": 6}, in: {...}}`

**Comment by alexandrasp** [comment]
Working on it.

**Comment by djmitche** [comment]
@alexandrasp I just noticed in the demo site,

```yaml
$let:
  ts: 100
  foo: 200
in:
  - $eval: ts+foo
  - $eval: ts-foo
  - $eval: ts*foo
```
fails.  The site is at https://taskcluster.github.io/json-e/

Can you take a look, add a test case, and fix it up?

**Comment by alexandrasp** [comment]
yes, @djmitche. I will include this case test and fix it.

**Comment by djmitche** [comment]
Fixed in #183
